File: apps/node-backend/src/services/redisClient.ts
import { Redis } from "ioredis";
import { env } from "../config/env";
import { logger } from "../lib/logger";

export const redisClient = new Redis(env.REDIS_URL);

redisClient.on("error", (err) => {
  logger.error({ err }, "Redis client error");
});

export async function checkRedisConnection(): Promise<boolean> {
  try {
    const pong = await redisClient.ping();
    return pong === "PONG";
  } catch (err) {
    logger.error({ err }, "Redis readiness check failed");
    return false;
  }
}

export async function closeRedisConnection(): Promise<void> {
  await redisClient.quit();
}

File: apps/ai-service/app/services/redis_client.py
import redis.asyncio as redis

from app.config import settings

_client: redis.Redis | None = None


def get_client() -> redis.Redis:
    global _client
    if _client is None:
        _client = redis.from_url(settings.redis_url, decode_responses=True)
    return _client


async def check_redis_connection() -> bool:
    try:
        client = get_client()
        return await client.ping()
    except Exception:
        return False


async def close_redis_connection() -> None:
    global _client
    if _client is not None:
        await _client.aclose()
        _client = None
